src/bin.ts

```diff
@@ -79,29 +79,71 @@ export function main (argv: string[]) {
     stopAtPositional: true
   })
 
-  const {
-    '--dir': dir = DEFAULTS.dir,
-    '--help': help = false,
-    '--script-mode': scriptMode = false,
-    '--version': version = 0,
-    '--require': requires = [],
-    '--eval': code = undefined,
-    '--print': print = false,
-    '--interactive': interactive = false,
-    '--files': files = DEFAULTS.files,
-    '--compiler': compiler = DEFAULTS.compiler,
-    '--compiler-options': compilerOptions = DEFAULTS.compilerOptions,
-    '--project': project = DEFAULTS.project,
-    '--ignore-diagnostics': ignoreDiagnostics = DEFAULTS.ignoreDiagnostics,
-    '--ignore': ignore = DEFAULTS.ignore,
-    '--transpile-only': transpileOnly = DEFAULTS.transpileOnly,
-    '--pretty': pretty = DEFAULTS.pretty,
-    '--skip-project': skipProject = DEFAULTS.skipProject,
-    '--skip-ignore': skipIgnore = DEFAULTS.skipIgnore,
-    '--prefer-ts-exts': preferTsExts = DEFAULTS.preferTsExts,
-    '--log-error': logError = DEFAULTS.logError,
-    '--emit': emit = DEFAULTS.emit
-  } = args
+  const flagOptions = {
```

Comment:
I'm _really_ not a fan of these changes, it's super opaque and confusing. This is a large reason I said we should probably change the ordering in a follow up PR, it's taking away from the actual feature. In general, I think we should revert all these changes and tackle it in a follow up PR. If you still want to tackle it, move `DEFAULTS` into `index.ts` _only_ and don't use them in `bin`. The current code would be better represented by just extending `options, tsconfig, DEFAULTS` where each option is previously `undefined`.

Reply:
I'll roll them back.  If I understand you correctly, the implementation you're proposing is to never pass defaults from bin.ts into `register()` nor from `register()` into `create()`?  Instead, read them only when actually used?

Reply:
Right, I believe we can read them only in `create()` if we use the `defaults()` function you created. This is because the values are all from `process.env` and should be undefined (or `null` in the case of `yn` being used).

Reply:
I'm upgrading yn to the latest version, which returns `undefined` instead of `null`.  This is good cuz it matches JS semantics for arg and destructuring defaults.